llvm/include/llvm/Target/TargetSelectionDAG.td

```diff
@@ -884,6 +884,9 @@ class PatFrags<dag ops, list<dag> frags, code pred = [{}],
   // If set to true, a predicate is added that checks for the absence of use of
   // the first result.
   bit HasNoUse = ?;
+  // If set to true, a predicate is added that checks for the sole use of
+  // the first result.
```

Comment:
```suggestion
  // If set to true, a predicate is added that checks for that the first result only has one non-debug user.
```

Reply:
dag emitter checks `hasOneUse()` though. This file is generic to both emitters.

Also, `GIM_CheckHasNoUse` is implemented with `use_nodbg_empty`, yet the comment above `bit HasNoUse = ?;` has non-debug users as implicit.

Reply:
I wouldn't bother spelling out the debug user case; it shouldn't be relevant much longer since debug instructions are going away

Reply:
Ok. Changed it back to something similar yet without mentioning that. thank you.